File: baekjoon/02500/2579.py
import sys
read = sys.stdin.readline

dp = [0] * 301
stairs = [0] * 301

n = int(read())
for i in range(1, n+1):
    stairs[i] = int(input())

dp[1] = stairs[1]; dp[2] = stairs[1] + stairs[2]
for i in range(3, n+1):
    dp[i] = max(dp[i-2], stairs[i-1] + dp[i-3]) + stairs[i]
print(dp[n])

File: baekjoon/02500/_2579.java
import java.io.*;

class Main {

    int[] dp = new int[301];

    private void init_dp(int[] arr) {
        dp[1] = arr[1];
        dp[2] = arr[1] + arr[2];
    }

    private void sol() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] stairs = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            stairs[i] = Integer.parseInt(br.readLine());
        }
        if (n == 1) {
            System.out.println(stairs[1]);
            return;
        }
        init_dp(stairs);

        for (int i = 3; i < n + 1; i++) {
            dp[i] = Math.max(stairs[i - 1] + dp[i - 3], dp[i - 2]) + stairs[i];
        }
        System.out.println(dp[n]);
    }

    public static void main(String[] args) throws Exception {
        new Main().sol();
    }
}